src/SoftwareSerial.h

```diff
@@ -32,13 +32,71 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+	SWSERIAL_5E1 = ( SWSER_NB_BIT_5 | SWSER_PARITY_EVEN  | SWSER_NB_STOP_BIT_1 ),
+	SWSERIAL_6E1 = ( SWSER_NB_BIT_6 | SWSER_PARITY_EVEN  | SWSER_NB_STOP_BIT_1 ),
+	SWSERIAL_7E1 = ( SWSER_NB_BIT_7 | SWSER_PARITY_EVEN  | SWSER_NB_STOP_BIT_1 ),
+	SWSERIAL_8E1 = ( SWSER_NB_BIT_8 | SWSER_PARITY_EVEN  | SWSER_NB_STOP_BIT_1 ),
+	SWSERIAL_5O1 = ( SWSER_NB_BIT_5 | SWSER_PARITY_ODD   | SWSER_NB_STOP_BIT_1 ),
+	SWSERIAL_6O1 = ( SWSER_NB_BIT_6 | SWSER_PARITY_ODD   | SWSER_NB_STOP_BIT_1 ),
+	SWSERIAL_7O1 = ( SWSER_NB_BIT_7 | SWSER_PARITY_ODD   | SWSER_NB_STOP_BIT_1 ),
+	SWSERIAL_8O1 = ( SWSER_NB_BIT_8 | SWSER_PARITY_ODD   | SWSER_NB_STOP_BIT_1 ),
+	SWSERIAL_5M1 = ( SWSER_NB_BIT_5 | SWSER_PARITY_MARK  | SWSER_NB_STOP_BIT_1 ),
+	SWSERIAL_6M1 = ( SWSER_NB_BIT_6 | SWSER_PARITY_MARK  | SWSER_NB_STOP_BIT_1 ),
+	SWSERIAL_7M1 = ( SWSER_NB_BIT_7 | SWSER_PARITY_MARK  | SWSER_NB_STOP_BIT_1 ),
+	SWSERIAL_8M1 = ( SWSER_NB_BIT_8 | SWSER_PARITY_MARK  | SWSER_NB_STOP_BIT_1 ),
+	SWSERIAL_5S1 = ( SWSER_NB_BIT_5 | SWSER_PARITY_SPACE | SWSER_NB_STOP_BIT_1 ),
+	SWSERIAL_6S1 = ( SWSER_NB_BIT_6 | SWSER_PARITY_SPACE | SWSER_NB_STOP_BIT_1 ),
+	SWSERIAL_7S1 = ( SWSER_NB_BIT_7 | SWSER_PARITY_SPACE | SWSER_NB_STOP_BIT_1 ),
+	SWSERIAL_8S1 = ( SWSER_NB_BIT_8 | SWSER_PARITY_SPACE | SWSER_NB_STOP_BIT_1 ),
+	SWSERIAL_5N2 = ( SWSER_NB_BIT_5 | SWSER_PARITY_NONE  | SWSER_NB_STOP_BIT_2 ),
+	SWSERIAL_6N2 = ( SWSER_NB_BIT_6 | SWSER_PARITY_NONE  | SWSER_NB_STOP_BIT_2 ),
+	SWSERIAL_7N2 = ( SWSER_NB_BIT_7 | SWSER_PARITY_NONE  | SWSER_NB_STOP_BIT_2 ),
+	SWSERIAL_8N2 = ( SWSER_NB_BIT_8 | SWSER_PARITY_NONE  | SWSER_NB_STOP_BIT_2 ),
+	SWSERIAL_5E2 = ( SWSER_NB_BIT_5 | SWSER_PARITY_EVEN  | SWSER_NB_STOP_BIT_2 ),
+	SWSERIAL_6E2 = ( SWSER_NB_BIT_6 | SWSER_PARITY_EVEN  | SWSER_NB_STOP_BIT_2 ),
+	SWSERIAL_7E2 = ( SWSER_NB_BIT_7 | SWSER_PARITY_EVEN  | SWSER_NB_STOP_BIT_2 ),
+	SWSERIAL_8E2 = ( SWSER_NB_BIT_8 | SWSER_PARITY_EVEN  | SWSER_NB_STOP_BIT_2 ),
+	SWSERIAL_5O2 = ( SWSER_NB_BIT_5 | SWSER_PARITY_ODD   | SWSER_NB_STOP_BIT_2 ),
+	SWSERIAL_6O2 = ( SWSER_NB_BIT_6 | SWSER_PARITY_ODD   | SWSER_NB_STOP_BIT_2 ),
+	SWSERIAL_7O2 = ( SWSER_NB_BIT_7 | SWSER_PARITY_ODD   | SWSER_NB_STOP_BIT_2 ),
+	SWSERIAL_8O2 = ( SWSER_NB_BIT_8 | SWSER_PARITY_ODD   | SWSER_NB_STOP_BIT_2 ),
+	SWSERIAL_5M2 = ( SWSER_NB_BIT_5 | SWSER_PARITY_MARK  | SWSER_NB_STOP_BIT_2 ),
+	SWSERIAL_6M2 = ( SWSER_NB_BIT_6 | SWSER_PARITY_MARK  | SWSER_NB_STOP_BIT_2 ),
+	SWSERIAL_7M2 = ( SWSER_NB_BIT_7 | SWSER_PARITY_MARK  | SWSER_NB_STOP_BIT_2 ),
+	SWSERIAL_8M2 = ( SWSER_NB_BIT_8 | SWSER_PARITY_MARK  | SWSER_NB_STOP_BIT_2 ),
+	SWSERIAL_5S2 = ( SWSER_NB_BIT_5 | SWSER_PARITY_SPACE | SWSER_NB_STOP_BIT_2 ),
+	SWSERIAL_6S2 = ( SWSER_NB_BIT_6 | SWSER_PARITY_SPACE | SWSER_NB_STOP_BIT_2 ),
+	SWSERIAL_7S2 = ( SWSER_NB_BIT_7 | SWSER_PARITY_SPACE | SWSER_NB_STOP_BIT_2 ),
+	SWSERIAL_8S2 = ( SWSER_NB_BIT_8 | SWSER_PARITY_SPACE | SWSER_NB_STOP_BIT_2 )
 };
 
+enum ParityMode : uint8_t { NONE = 0, ODD, EVEN, SPACE, MARK };
+
```

Comment:
No surrounding namespace, not class member enum, hence please use unique identifiers style, SWSERIAL_PARITY_.... or the likes.

Reply:
Also spelled out SWSER to SWSERIAL and added _CONF to the bitfield names to more clearly separate them from the ParityMode enum. I guess the whole block would benefit (at least visually) from shorter/simpler names but I'm out of ideas for now.